### ISSUE TITLE
Document how to set up a Gitter-like user experience

### ISSUE THREAD

**Issue by pdurbin** [modernxmpp/modernxmpp#21]
Gitter is a fantastic discussion platform for open source projects because discussion and decision making can take place in the open. The community can catch up on important moments in the chat history by navigating to the permalink in the archive.

Let's use https://gitter.im/good-labs as an example. Can the user experience be replicated using XMPP?

Step 1: Navigate to https://gitter.im/good-labs and click a room, such as "community".

<img width="1422" alt="Screen Shot 2019-05-02 at 8 49 29 AM" src="https://user-images.githubusercontent.com/21006/57076275-42615200-6cb7-11e9-8adb-417f460b1045.png">

Step 2: Catch up on the chatter

<img width="1449" alt="Screen Shot 2019-05-02 at 8 51 39 AM" src="https://user-images.githubusercontent.com/21006/57076383-881e1a80-6cb7-11e9-9100-bde35c2a6010.png">

Step 3: Click "Sign in to start talking".

<img width="1256" alt="Screen Shot 2019-05-02 at 8 52 56 AM" src="https://user-images.githubusercontent.com/21006/57076463-b996e600-6cb7-11e9-8bf3-13b1243b32e3.png">

I understand that some XMPP rooms are logged such as http://logs.xmpp.org/xsf but when I am catching up on the chatter at http://logs.xmpp.org/xsf/2019-05-02?p=h for example, there is no "Sign in to start talking" button. Here's a screenshot:

<img width="1422" alt="Screen Shot 2019-05-02 at 8 54 30 AM" src="https://user-images.githubusercontent.com/21006/57076623-1397ab80-6cb8-11e9-859c-8089408e7fd3.png">

I **think** the first step is to take an existing XMPP log page and add a "Sign in to start talking" button. Does that make sense? Thanks.

**Comment by mwild1** [comment]
The top of that logs page has two links: "Join using a client" and "Join via web" which both do what you want really. The issue probably comes down to the design not making them obvious. It was produced by developers, not designers :)

That specific page is generated by Prosody, and discussion about changing its design would obviously be better discussed within that project.

This project (Modern XMPP) is about providing advice and guidelines to developers of XMPP software. So if we want to document "These are the key things to consider when designing an XMPP room logs page" then that is appropriate here.

**Comment by pdurbin** [comment]
@mwild1 duh. You're right. Those two "join" links are right at the top. Maybe they aren't obvious enough, not enough of a call to action. This is reminding me that people are confused when visiting IRC logs at http://irclog.iq.harvard.edu/dataverse/2019-05-02 for example (screenshot below) for a similar reason, perhaps. I wrote "Connect via [chat.dataverse.org](http://chat.dataverse.org)..." but maybe I'll change it to a big orange "Sign it to start talking" button to be more like Gitter. 😄 

<img width="1034" alt="Screen Shot 2019-05-02 at 9 17 16 AM" src="https://user-images.githubusercontent.com/21006/57077974-573fe480-6cbb-11e9-8471-3257789e6127.png">

I'm glad you think the idea is appropriate. Please let me know how I can help!

**Comment by cal0pteryx** [comment]
For prosody, I guess this is the relevant module?

https://modules.prosody.im/mod_muc_log_http.html

I see it should have theming support as well.

**Comment by pdurbin** [comment]
@wurstsalat3000 yes, I believe that's the right one, in use at http://logs.xmpp.org/xsf/ which I linked earlier and https://conference.gajim.org:5281/muc_log/gajim/ and probably lots of places.

**Comment by pdurbin** [comment]
Over at https://github.com/good-labs/slopi-communication/issues/2 I'm starting to make a list of examples of what I'm calling SLOPI chat.

SLOPI is pronounced "sloppy" and means Searchable Linkable Open Public Indexed (SLOPI) chat. The only XMPP example I have so far is http://logs.xmpp.org/xsf/ but maybe I'll add Gajim to the list.

I'd love feedback on the idea of SLOPI. :smile:

**Comment by pdurbin** [comment]
Today I've been talking out a lot of my requirements at https://logs.xmpp.org/xsf/2019-06-09?p=h

@poikilotherm I've been trying to represent some of your requirements from https://github.com/IQSS/chat.dataverse.org/issues/7 as well. :smile:

**Comment by pdurbin** [comment]
I recently implemented a change to http://chat.dataverse.org to show a preview of recent messages. Here's how it looks:

<img width="1149" alt="Screen Shot 2019-09-29 at 8 03 21 AM" src="https://user-images.githubusercontent.com/21006/65832286-a2bce580-e28f-11e9-8947-02127c77c67b.png">

Obviously, this is not nearly as good as what Gitter offers. For my implementation, you can't scroll back at all. You are stuck staring at the three bullets, the three most recent messages. And as people talk, the bullets are not replaced. You have to refresh the page to see new messages. All this is to say that my approach is not what I want for XMPP but I hope it helps illustrates what I want, which is a preview of messages right on the "home page" for the chat room (just like Gitter).

**Comment by poikilotherm** [comment]
It would be really cool to include the date of the messages... :smile:

**Comment by poVoq** [comment]
See a similar idea for ConverseJS:
https://github.com/conversejs/community-plugins/issues/4
It has been implemented for Wikisuits, but never backported into upstream ConverseJS I think.

Edit: You can try it here: https://wikisuite.chat/XMPP

**Comment by pdurbin** [comment]
@poVoq WOW! THANKS! I agree that https://wikisuite.chat/XMPP is pretty much exactly what I'm talking about. Here's a screenshot:

<img width="1174" alt="Screen Shot 2019-11-14 at 8 01 36 PM" src="https://user-images.githubusercontent.com/21006/68908553-b629f000-0719-11ea-89ab-9c24ade92014.png">

It looks like https://github.com/conversejs/converse.js/pull/1266 hasn't been merged though. Bummer!

I just added WikiSuite to my "Examples of communities that use the Searchable Linkable Open Public Indexed (SLOPI) communication style" spreadsheet at https://docs.google.com/spreadsheets/d/1wvG3XTd5YwA-SliOUCavQgqfK1jLVnlL9tuUsjHR0Ik/edit?usp=sharing 😄 

I wasn't easily able to figure out where the message archives are for WikiSuite ("Message archiving" under "Features isn't clickable) but I assume they're somewhere! Thanks!

**Comment by Zash** [comment]
> It looks like conversejs/converse.js#1266 hasn't been merged though. Bummer!

Looks like it has now! [Docs](https://conversejs.org/docs/html/configuration.html#muc-show-logs-before-join)
